### PR TITLE
v1.18.0

### PR DESCRIPTION
**Pull request by jmjurado23**
# Feature/add new rules v1.18.0

- New gems:
   - rubocop-factory_bot: 2.24.0
   
- Update gems:

   - rubocop: 1.48.1 => 1.57.1
   - rubocop-capybara: 2.17.1 => 2.19.0
   - rubocop-performance: 1.16.0 => 1.19.1
   - rubocop-rails: 2.18.0 => 2.21.2
   - rubocop-rspec: 2.21.2 => 2.24.1
   
 - Added new rules
 
    - rubocop
   
      - Style/RedundantLineContinuation (1.49.0)
      - Style/DataInheritance (1.49.0)
      - Lint/DuplicateMatchPattern (1.50.0)
      - Style/ExactRegexpMatch (1.51.0)
      - Style/RedundantArrayConstructor (1.52.0)
      - Style/RedundantRegexpConstructor (1.52.0)
      - Style/RedundantFilterChain (1.52.0)
      - Lint/MixedCaseRange (1.53.0)
      - Lint/RedundantRegexpQuantifiers (1.53.0)
      - Style/RedundantCurrentDirectoryInPath (1.53.0)
      - Style/ReturnNilInPredicateMethodDefinition (1.53.0)
      - Style/YAMLFileRead (1.53.0)
      - Bundler/DuplicatedGroup (1.56.0)
      - Style/SingleLineDoEndBlock (1.57.0)
      
    - rubocop-capybara
    
      - Capybara/RSpec/PredicateMatcher (2.19.0)
      - Capybara/RSpec/HaveSelector (2.19.0)
      - Capybara/ClickLinkOrButtonStyle (2.19.0)
      
    - rubocop-performance
    
      - Performance/MapMethodChain (1.19.0)
      
    - rubocop-rails
    
      - Rails/ThreeStateBooleanColumn (2.19.0)
      - Rails/SelectMap (2.21.0)
      - Rails/DangerousColumnNames (2.21.0)
      - Rails/UnusedRenderContent (2.21.0)
      - Rails/RedundantActiveRecordAllMethod (2.21.0)
      
    - rubocop-rspec
    
      - RSpec/IndexedLet (2.20.0)
      - RSpec/BeEmpty (2.20.0)
      - RSpec/Rails/NegationBeValid (2.23.0)
      - RSpec/ReceiveMessages (2.23.0)
      - RSpec/EmptyMetadata (2.24.0)
      - RSpec/Eq (2.24.0)
      - RSpec/MetadataStyle (2.24.0)
      - RSpec/SpecFilePathFormat (2.24.0)
      - RSpec/SpecFilePathSuffix (2.24.0)
      
    - rubocop-factory_bot
    
      - FactoryBot/AssociationStyle (2.23.0)
      - FactoryBot/FactoryAssociationWithStrategy (2.23.0)
      - FactoryBot/IdSequence (2.23.0)
      - FactoryBot/RedundantFactoryOption (2.23.0)

### CODE REVIEW

File: rubocop-capybara.yml [hunk 1]
```diff
@@ -24,3 +24,16 @@ Capybara/SpecificActions:
...
+  Enabled: true
+
+# https://docs.rubocop.org/rubocop-capybara/cops_capybara_rspec.html#capybararspechaveselector
+Capybara/RSpec/HaveSelector:
+  Enabled: true
+
+# https://docs.rubocop.org/rubocop-capybara/cops_capybara.html#capybaraclicklinkorbuttonstyle
+Capybara/ClickLinkOrButtonStyle:
+  Enabled: true
+  EnforcedStyle: link_or_button
```
Comment (pacop) [inline rubocop-capybara.yml:39]: Como este es el valor por defecto quizás podríamos omit esta linea(39)?

File: rubocop-factory-bot.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+require: rubocop-factory_bot
+
+# https://docs.rubocop.org/rubocop-factory_bot/cops_factorybot.html#factorybotsyntaxmethods
+FactoryBot/SyntaxMethods:
+  Enabled: true
+
+# https://docs.rubocop.org/rubocop-rspec/cops_rspec_factorybot.html#rspecfactorybotconsistentparenthesesstyle
+FactoryBot/ConsistentParenthesesStyle:
+  Enabled: true
+  EnforcedStyle: "require_parentheses"
```
Comment (pacop) [inline rubocop-factory-bot.yml:14]: Siguiendo la misma filosofia del anterior comentario, quizas podamos quitar esta linea tambien por ser el de por defecto?

**Review by pacop — APPROVED** [review]

File: rubocop-capybara.yml (hunk 1 above)
Comment (jmjurado23) [reply rubocop-capybara.yml]: ok

File: rubocop-factory-bot.yml (hunk 2 above)
Comment (jmjurado23) [reply rubocop-factory-bot.yml]: ok

**Pull request merged**